### PR TITLE
support for Vim (in terminal) on OS X

### PR DESCRIPTION
**Pull request by pofigizm**

### CODE REVIEW

**Comment by lahmatiy** [comment]
Wow! Awesome!
Thank you 👍

File: lib/editors/vim.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+var Promise = require('es6-promise-polyfill').Promise;
+
+var osascript = function(script) {
+  return 'osascript -e \'' + script + '\'';
+}
```
Comment (lahmatiy) [inline lib/editors/vim.js]: Semicolon please

File: lib/editors/vim.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+var terminal = function(cmd) {
+  return 'tell application "Terminal" to do script "' + cmd + '"';
+}
```
Comment (lahmatiy) [inline lib/editors/vim.js]: Semicolon please

File: lib/editors/vim.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+var commands = function() {
+  var cmds = [].slice.call(arguments);
+  return cmds.join('; ');
+}
```
Comment (lahmatiy) [inline lib/editors/vim.js]: Semicolon please

File: lib/utils.js [hunk 4]
```diff
@@ -5,12 +5,14 @@ module.exports = {
   number: function(value, fallback) {
     return isNaN(value) ? fallback : value;
   },
-  quote: function(value) {
+  quote: function(value, escape) {
```
Comment (lahmatiy) [inline lib/utils.js]: `escape` -> `escapeQuotes`

File: lib/editors/vim.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+var osascript = function(script) {
+  return 'osascript -e \'' + script + '\'';
+}
+
+var terminal = function(cmd) {
+  return 'tell application "Terminal" to do script "' + cmd + '"';
+}
+
+var commands = function() {
+  var cmds = [].slice.call(arguments);
```
Comment (lahmatiy) [inline lib/editors/vim.js]:
```js
return [].join.call(arguments, '; ');
```

File: lib/editors/vim.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  onlyPattern: true,
+  escapeQuotes: true,
+  pattern: osascript(
+    terminal(
+      commands(
+        'cd {projectPath}',
+        'vim {filename} \\"+call cursor({line}, {column})\\"'
+      )
+    )
+  ),
```
Comment (lahmatiy) [inline lib/editors/vim.js]: No comma please

File: lib/editors/vim.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  return 'tell application "Terminal" to do script "' + cmd + '"';
+}
+
+var commands = function() {
+  var cmds = [].slice.call(arguments);
+  return cmds.join('; ');
+}
+
+var settings = {
+  onlyPattern: true,
```
Comment (lahmatiy) [inline lib/editors/vim.js]: maybe `patternOnly`?

**Review by lahmatiy — CHANGES_REQUESTED** [review]

File: lib/editors/vim.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+var osascript = function(script) {
+  return 'osascript -e \'' + script + '\'';
+};
+
+var terminal = function(cmd) {
+  return 'tell application "Terminal" to do script "' + cmd + '"';
+};
+
+var commands = function() {
+  return [].join.call(arguments, '; ');
```
Comment (lahmatiy) [inline lib/editors/vim.js]: Looked again. Are you sure we need that function? ;)

Comment (pofigizm) [reply lib/editors/vim.js]: http://pastebin.com/b76rsvux
?

Comment (lahmatiy) [reply lib/editors/vim.js]: Why not make it simple?
```js
'cd {projectPath}; ' +
'vim {filename} \\"+call cursor({line}, {column})\\"'
```

Comment (pofigizm) [reply lib/editors/vim.js]: But I think call place became less readable.

Comment (pofigizm) [reply lib/editors/vim.js]: )) ok

**Review by lahmatiy — APPROVED** [review]

**Comment by tyanas** [comment]
❤️  ❤️  😍

**Comment by lahmatiy** [comment]
@pofigizm Thank you! 👍

**Pull request merged**